nvflare/private/fed/client/client_executor.py

```diff
@@ -360,16 +359,16 @@ def _wait_child_process_finish(self, client, run_number, allocated_resource, tok
             if time.time() - start > 15:
                 break
 
-        self.logger.info("waiting for process to finish.")
+        self.logger.info(f"run ({run_number}): waiting for child worker process to finish.")
         with self.lock:
             child_process = self.run_processes.get(run_number, {}).get(RunProcessKey.CHILD_PROCESS)
         if child_process:
             child_process.wait()
             return_code = child_process.returncode
-            self.logger.info(f"process finished with execution code: {return_code}")
+            self.logger.info(f"run ({run_number}): child worker process finished with execution code: {return_code}")
 
         if allocated_resource:
-            resource_manager.free_resources(resources=allocated_resource, token=token, fl_ctx=FLContext())
+            resource_manager.free_resources(resources=allocated_resource, token=token)
```

Comment:
Here, instead of using:
 fl_ctx=FLContext()

we can change it to:
fl_ctx = client.engine.new_context()